src/types/instruction.rs

```diff
@@ -124,4 +145,29 @@ mod tests {
             decode_instruction(encoded_instruction.to_i64().unwrap(), Some(&Felt::new(2))).unwrap();
         assert_eq!(instruction.size(), 2);
     }
+
+    #[test]
+    fn test_instruction_display() {
+        let test_instruction = Instruction {
+            off0: 1,
+            off1: 2,
+            off2: 3,
+            imm: Some(Felt {
+                value: FeltBigInt {
+                    val: BigUint { data: vec![0] },
+                },
+            }),
+            dst_register: Register::AP,
+            op0_register: Register::AP,
+            op1_addr: Op1Addr::Imm,
+            res: Res::Add,
+            pc_update: PcUpdate::Regular,
+            ap_update: ApUpdate::Regular,
+            fp_update: FpUpdate::Regular,
+            opcode: Opcode::Call,
+        };
+
+        let expected_output = "off0: 1\noff1: 2\noff2: 3\nimm: Some(Felt { value: FeltBigInt { val: BigUint { data: [0] } } })\ndst_register: AP\nop0_register: AP\nop1_addr: Imm\nres: Add\npc_update: Regular\nap_update: Regular\nfp_update: Regular\nopcode: Call\n";
```

Comment:
This exposes way too much data about internal representation. It looks like we just want the `Debug` trait to me.